src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorPickerSlider.cs

```diff
@@ -0,0 +1,283 @@
+﻿using System;
+using Windows.System;
+using Windows.UI;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Uno.UI.Helpers.WinUI;
+
+namespace Microsoft.UI.Xaml.Controls.Primitives
+{
+	public partial class ColorPickerSlider : Slider
+	{
+		ToolTip m_toolTip;
+
+		public ColorPickerSlider()
+		{
+			// We want the ColorPickerSlider to pick up everything for its default style from the Slider's default style,
+			// since its purpose is just to turn off keyboarding.  So we'll give it Slider's control name as its default style key
+			// instead of ColorPickerSlider.
+			this.DefaultStyleKey = typeof(Windows.UI.Xaml.Controls.Slider);
+
+			this.ValueChanged += OnValueChangedEvent;
+		}
+
+		// IUIElementOverridesHelper overrides
+		protected override AutomationPeer OnCreateAutomationPeer()
+		{
+			return new ColorPickerSliderAutomationPeer(this);
+		}
+
+		// IFrameworkElementOverrides overrides
+		protected override void OnApplyTemplate()
+		{
+			base.OnApplyTemplate();
+
+			m_toolTip = this.GetTemplateChild<ToolTip>("ToolTip");
+
+			if (m_toolTip is ToolTip toolTip)
+			{
+				toolTip.Content = this.GetToolTipString();
+			}
+		}
+
+		// IControlOverrides overrides
+		protected override void OnKeyDown(KeyRoutedEventArgs args)
+		{
+			if ((this.Orientation == Orientation.Horizontal &&
+						 args.Key != VirtualKey.Left &&
+						 args.Key != VirtualKey.Right) ||
+				(this.Orientation == Orientation.Vertical &&
+						 args.Key != VirtualKey.Up &&
+						 args.Key != VirtualKey.Down))
+			{
+				base.OnKeyDown(args);
+				return;
+			}
+
+			ColorPicker parentColorPicker = GetParentColorPicker();
+
+			if (parentColorPicker == null)
+			{
+				return;
+			}
+
+			bool isControlDown = (Window.Current.CoreWindow.GetKeyState(VirtualKey.Control) & CoreVirtualKeyStates.Down) == CoreVirtualKeyStates.Down;
+
+			double minBound = 0;
+			double maxBound = 0;
+
+			Hsv currentHsv = parentColorPicker.GetCurrentHsv();
+			double currentAlpha = 0;
+
+			switch (this.ColorChannel)
+			{
+				case ColorPickerHsvChannel.Hue:
+					minBound = parentColorPicker.MinHue;
+					maxBound = parentColorPicker.MaxHue;
+					currentHsv.H = this.Value;
+					break;
+
+				case ColorPickerHsvChannel.Saturation:
+					minBound = parentColorPicker.MinSaturation;
+					maxBound = parentColorPicker.MaxSaturation;
+					currentHsv.S = this.Value / 100;
+					break;
+
+				case ColorPickerHsvChannel.Value:
+					minBound = parentColorPicker.MinValue;
+					maxBound = parentColorPicker.MaxValue;
+					currentHsv.V = this.Value / 100;
+					break;
+
+				case ColorPickerHsvChannel.Alpha:
+					minBound = 0;
+					maxBound = 100;
+					currentAlpha = this.Value / 100;
+					break;
+
+				default:
+					throw new Exception(); // Uno Doc: 'throw winrt::hresult_error(E_FAIL);'
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)